CMakeLists.txt

```diff
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.1.3 FATAL_ERROR)
+project(hidapi LANGUAGES C)
+
+if(NOT DEFINED CMAKE_BUILD_TYPE OR CMAKE_BUILD_TYPE STREQUAL "")
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Debug;Release;MinSizeRel;RelWithDebInfo" FORCE)
+endif()
+set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+
```

Comment:
so instead of type less, you suggest always type more (and possible type more errors)
I vote for user-friendly

> muscle memory for me personally

and it makes it harder to use for those, who don't have such muscle memory

---
even more: I use cmake-gui even more often then a command-line. It saves me much time. And it saves me time to pick a selection from a combo-box, instead of typing it manually each time.

Reply:
> so instead of type less, you suggest always type more

In this case, yes. It's not about typing less/more. It's about not colliding with CMake builtins in the future.

Further, most people that will be using this will be either installing from a package manager or using it as a submodule - and only the latter would require someone typing it once to configure it.

> and it makes it harder to use for those, who don't have such muscle memory

The point wasn't that I'm better. The point is that it's a pretty standard way to use CMake.

> even more: I use cmake-gui even more often then a command-line

That is not the majority case.

Reply:
> or using it as a submodule - and only the latter would require someone typing it once to configure it.

And what problem does it make? Does it limit the users?

> The point wasn't that I'm better. 

Neither was mine.

> The point is that it's a pretty standard way to use CMake.

For all my "small" project I have a different muscle memory: `cmake .. && make -j`. Pretty standard to me (and I didn't invented it).

Reply:
The point here is that this is not common. Another user now has asked about it; you generally don't set these things, because debug/release isn't binary. I've rarely, hardly ever actually, seen a config that does this. It's the general case yo specify the config type you want to build for.